imap/jmap_contact.c

```diff
@@ -863,6 +863,34 @@ static int _add_othergroup_entries(struct jmap_req *req,
     return r;
 }
 
+typedef struct {
+    const char *key;
+    const char *prop;
+    char *type;
+    size_t size;
+    struct message_guid guid;
+} property_blob_t;
+
+static property_blob_t *property_blob_new(const char *key, const char *prop,
+                                          char *type, struct buf *data)
+{
+    property_blob_t *blob = xzmalloc(sizeof(property_blob_t));
+
+    blob->key = key;
+    blob->prop = prop;
+    blob->type = type;
```

Comment:
I'd suggest not to transfer ownership here, it's not really clear from the function signature. Since the only caller of property_call_new passes a xstrdup'ed copy, I think that type should become a const pointer and the xstrdup is done within the function.